hashintel/hash
# Make GraphQL API runnable, basic fixes in `Entity`/`EntityType`/`PropertyType`/`DataType` model classes

## 🌟 What is the purpose of this PR?




This PR makes the GraphQL API runnable again, by removing references to the deprecated link type model, and refactoring the `Entity`/`EntityType`/`PropertyType`/`DataType` model classes.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203150072758509/1203329839349722/f) _(internal)_
- [Update to the new Type System package in the Workspace](https://app.asana.com/0/1201095311341924/1203259817761581/f)

## 🔍 What does this change?




See commits.

## 🐾 Next steps



- [Replace link types with entity types in the HASH App on the `dev/links` branch](https://app.asana.com/0/0/1203361844133477/f)
- [Fix links in the HASH App on the `dev/links` branch](https://app.asana.com/0/0/1203361844133479/f)
- [Replace uses of `entityId` in the codebase with `entityUuid` and `baseId`
](https://app.asana.com/0/0/1203361740074435/f)
- [Rework the TS/GraphQL types to be consistent across the codebase](https://app.asana.com/0/1201095311341924/1203227079758117/f)

## ❓ How to test this?



1. Checkout the branch
2. Run the external services using `yarn external-services up`
3. Run the GraphApi using `yarn dev` in a separate terminal window
4. Go to `localhost:5001/graphql` to see the running GQL playground
